Python/wrfrun/fixIM.py

```diff
@@ -0,0 +1,151 @@
+SEAICE_fill_missing = -1E30 # Value to use to fill SEAICE missing points.
+correct_stlatlon = False # Whether or not to correct the stlat and stlon values. 
+# NOTE: To get stlat and stlon correction, "REPLACE_IM_STLATSTLON" environment variable
+#   needs to be set beforehand.
+sl_layers = ['000007','007028','028100','100289'] # Soil layers (cm-cm).
+plevs = np.array([                                                             
+  100000.0, 97500.0, 95000.0, 92500.0, 90000.0, 87500.0, 85000.0, 82500.0,               
+  80000.0,  77500.0, 75000.0, 70000.0, 65000.0, 60000.0, 55000.0, 50000.0, 
+  45000.0,  40000.0, 35000.0, 30000.0, 25000.0, 22500.0, 20000.0, 17500.0, 
+  15000.0,  12500.0, 10000.0, 7000.0,  5000.0,  3000.0,  2000.0,  1000.0, 
+  700.0,    500.0,   300.0,   200.0,   100.0]) # Grib pressure levels (Pa).
+# NOTE: sl_layers and plevs are default values that come with ERA5.  
+  
+  
+# ===============================================================================
+# ================================= Main program ================================
+# ===============================================================================
+    
+if __name__ == '__main__':
+
+  # Retreive date and pl and sl grib files' and IM file's names (and paths)  
+  IMDate = argv[1]
+  Grb_pl = argv[2]
+  Grb_sl = argv[3]
+  IMF = argv[4]
+  
+  # Check for "REPLACE_IM_STLATSTLON" environment variable  
+  if ('REPLACE_IM_STLATSTLON' in os.environ): 
+    stlatlon = os.environ['REPLACE_IM_STLATSTLON']  
+    sp = stlatlon.split(',')
+    if (len(sp) != 2):
+      raise ValueError('REPLACE_IM_STLATSTLON is not in the expected format')
+    stlat_a = float(sp[0]) 
+    stlon_a = float(sp[1])   
+    correct_stlatlon = True    
+  # NOTE: If present, "REPLACE_IM_SLATSLON" is expected to be of the form 'x,y', where  
+  #   x and y are the stlat_a and stlon_a values in degrees.  
+  
+  # Prompt on screen
+  print('\n================================== Processing ' + IMDate + ' ==================================')
+  
+  # Open IM file
+  interf = pyw.rinter(IMF)
+    
+  # Display keys
+  print('\nInput IM keys =',interf.keys())  
+  
+  # Get IM SEAICE values
+  IM_SEAICE = interf['SEAICE'].val
+  
+  # Open grib SL file & read seaice values
+  grbs = pygrib.open(Grb_sl)
+  grb = grbs.select(name='Sea ice area fraction')[0]
+  GRB_SEAICE = grb.values
+  
+  # Check that IM_SEAICE and GRB_SEAICE arrays are the same shape
+  if (IM_SEAICE.shape!=GRB_SEAICE.shape):
+    raise ValueError("IM_SEAICE.shape!=GRB_SEAICE.shape. "
+      "Intermediate and grib SEAICE arrays should be the same size.") 
+  
+  # Retreive stalt and stlon
+  stlat = interf['UU'].geoinfo['STARTLAT']
+  stlon = interf['UU'].geoinfo['STARTLON']
+  dlat = interf['UU'].geoinfo['DELTALAT']
+  dlon = interf['UU'].geoinfo['DELTALON']   
+  # NOTE: Here we assume all fields have the same stlat, etc.
+  
+  # Fix stlat and stlon values, if needed 
+  if (correct_stlatlon == True):
```

Comment:
you can just do: `if correct_stlatlon:`, since the variable by itself already evaluates to True; this also works with numbers and other objects, where zero or None are False and the rest is True, but that is more error-prone; for booleans, though, I think it makes things very readable (if they have an intuitive name)

Reply:
Done.